server/ENDPOINTS.md

```diff
@@ -0,0 +1,225 @@
+- If there are no errors, returned status is `OK`
+- Endpoints source code: `server/src/gateways/judges.gateway.ts`
+
+## `join` (work in progress) 
+
+Endpoint for judges to join the fight. Only previously selected judges can join the fight (they are identified by their ID).
+Endpoint checks if judge is allowed to join this fight, saves its socket and returns information about players.
+Players have no points, there are no events in the events' history, timer is set to zero minutes.
+
+### Parameters
+| name    | type   |
+|---------|--------|
+| fightId | string |
+| judgeId | string |
+
+### Response (send to the judge that called this endpoint)
+| name       | type   |
+|------------|--------|
+| redPlayer  | Player |
+| bluePlayer | Player |
+
+where `Player` is an interface:
+
+| name      | type   |
+|-----------|--------|
+| id        | string |
+| firstName | string |
+| lastName  | string |
+
+### Error codes
+- `NOT_FOUND` - fight with given fightId doesn't exist
+- `BAD_REQUEST` - judge with given judgeId has already joined fight with different socket
+- `UNAUTHORIZED` - judge with given judgeId is not allowed to join this fight
+
+## `startFight`
+
+Endpoint to start fight. Only main judge is allowed to start fight. Endpoint makes sure that 
+it is called by main judge, starts fight (changes fight state to `RUNNING` and starts timer) and 
+sends message to all judges.
+
+### Parameters
+| name    | type   |
+|---------|--------|
+| fightId | string |
+| judgeId | string |
+
+### Response (send to all judges)
+| name       | type   |
+|------------|--------|
+| status     | string |
+
+### Error codes
+- `NOT_FOUND` - fight with given fightId doesn't exist
+- `UNAUTHORIZED` - judge with given judgeId is not the main judge
+- `NOT_READY` - not all judges have already joined this fight
+- `BAD_REQUEST` - fight has already been started
+
+## `finishFight`
+
+Endpoint to finish fight. Only main judge is allowed to finish fight (even if the end conditions are not fulfilled). 
+Endpoint makes sure that it is called by main judge, finishes fight (changes fight state to `FINISHED` and stops timer) 
+and sends message to all judges.
+
+### Parameters
+| name    | type   |
+|---------|--------|
+| fightId | string |
+| judgeId | string |
+
+### Response (send to all judges)
+| name       | type   |
+|------------|--------|
+| status     | string |
+
+### Error codes
+- `NOT_FOUND` - fight with given fightId doesn't exist
+- `UNAUTHORIZED` - judge with given judgeId is not the main judge
+- `BAD_REQUEST` - fight has not been started or has already been finished
+
+## `pauseTimer`
+
+Endpoint to pause fight. Only main judge is allowed to pause fight. Endpoint makes sure that it is called by main judge, 
+pauses fight (changes fight state to `PAUSED` and pauses timer) and sends message to all judges.
+Additional parameter `exactPauseTimeInMillis` can be used to synchronize time between judges and server.
```

Comment:
Usunąłbym to additional, bo sugeruje, że exactPauseTimeInMillis jest opcjonalne, ale jest potrzebne zwłaszcza, jeśli fight ma ustawiony Timer ze względu na czasowy warunek zakończenia, a dodatkowo pozostali sędziowie mogą synchronizować u siebie czas zatrzymania, żeby zgadzał się z czasem zatrzymania u sędziego głównego